addons/equipviewer/equipviewer.lua

```diff
@@ -0,0 +1,734 @@
+    hide_on_zone = true,
+    hide_on_cutscene = true,
+    left_justify = false,
+}
+settings = nil
+local last_encumbrance_bitfield = 0
+
+-- gets the currently equipped item data for the slot information provided
+local function get_equipped_item(slotName, slotId, bag, index)
+    if not bag or not index then -- from memory
+        local equipment = windower.ffxi.get_items().equipment
+        bag = equipment[string.format('%s_bag', slotName)]
+        index = equipment[slotName]
+        if equipment_data[slotId] then
+            equipment_data[slotId].bag_id = bag
+            equipment_data[slotId].index = index
+        end
+    end
+    if index == 0 then -- empty equipment slot
+        return 0, 0
+    end
+    local item_data = windower.ffxi.get_items(bag, index)
+    return item_data.id, item_data.count
+end
+
+-- desc: Updates the ui object(s) for the given slot
+local function update_equipment_slot(source, slot, bag, index, item, count)
+    local slot_data = equipment_data[slot]
+    slot_data.bag_id = bag or slot_data.bag_id
+    slot_data.index = index or slot_data.index
+    if not item then
+        item, count = get_equipped_item(slot_data.slot_name, slot_data.slot_id, bag, index)
+    end
+    if evdebug then
+        bag = slot_data.bag_id
+        index = slot_data.index
+        log('%s %s %d %d %d':format(source, slot_data.slot_name, item, bag or -1, index or -1))
+        print('%s %s %d %d %d':format(source, slot_data.slot_name, item, bag or -1, index or -1))
+    end
+    if slot_data.slot_name  == 'ammo' then
+        slot_data.count = count or slot_data.count or 0
+    end
+    if slot_data.image and item ~= nil then
+        if item == 0 or item == 65535 then -- empty slot
+            slot_data.image:hide()
+            slot_data.image:clear()
+            slot_data.item_id = 0
+            slot_data.count = nil
+            slot_data.image:update()
+        elseif slot_data.item_id ~= item then
+            slot_data.item_id = item
+            local icon_path = string.format('%sicons/%s.bmp', windower.addon_path, slot_data.item_id)
+
+            if not windower.file_exists(icon_path) then
+                icon_extractor.item_by_id(slot_data.item_id, icon_path)
+            end
+            if windower.file_exists(icon_path) then
+                slot_data.image:path(icon_path)
+                slot_data.image:alpha(settings.icon.alpha)
+                slot_data.image:show()
+            end
+        end
+        if slot_data.slot_name == 'ammo' then
+            display_ammo_count(slot_data.count)
+        end
+        slot_data.image:update()
+    end
+end
+
+-- Updates the texture for all slots if it's a different piece of equipment
+local function update_equipment_slots(source)
+    for slot in pairs(equipment_data) do
+        update_equipment_slot(source, slot)
+    end
+end
+
+-- Sets up the image and text ui objects for our equipment
+local function setup_ui()
+    refresh_ui_settings()
+    destroy()
```

Comment:
Oh uh, I think I just didn't change them from local from the base code I originally worked on. I have no problem making all of them global if you're okay with that

Reply:
If anything I'd prefer you'd make the others local as well 😅 But can also just leave it as it is. I just thought maybe you made the others global because it didn't work otherwise.